tests/with_thiserror_tests.rs

```diff
@@ -0,0 +1,70 @@
+use anyhow::anyhow;
+use displaydoc::Display;
+use std::error::Error as _;
+use std::io;
+use thiserror::Error;
+
+fn assert_display<T: std::fmt::Display>(input: T, expected: &'static str) {
+    let out = format!("{}", input);
+    assert_eq!(expected, out);
+}
+
+#[test]
+fn test_transparent_for_enum() {
+    #[derive(Display, Error, Debug)]
+    enum MyError {
+        /// Doc for Variant.
+        #[display(transparent)]
+        Variant(anyhow::Error),
+    }
+
+    let var = MyError::Variant(anyhow!("inner").context("outer"));
+    assert_display(&var, "outer");
+    // assert_eq!(var.source().unwrap().to_string(), "inner")
+}
+
+#[test]
+fn test_transparent_for_struct() {
+    #[derive(Display, Error, Debug)]
+    #[display(transparent)]
```

Comment:
error.source() will work here if you do `#[error(transparent)]` and just let thiserror do the deriving for display.

Reply:
I commented in this tests and made sure its working.

```
error.source().unwrap().downcast_ref::<io::Error>().unwrap();
```

Probably because I have changed ```#[display(transparent)]```

```
#[error(transparent)]
```

you mentioned